src/Common/cmake/secure_dependencies.cmake

```
@@ -0,0 +1,73 @@
+#===============================================================================
+# Configure the flags used to ensure that required security settings are enabled
+macro(configure_security_flags)
+    # Enable Control Flow Guard
+    if (WIN32)
+        add_link_options("LINKER:/guard:cf")
+    endif()
+
+    # Enable Compiler supported Spectre mitigations
+    if (NOT APPLE)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mspeculative-load-hardening -mretpoline")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mspeculative-load-hardening -mretpoline")
+    endif()
+endmacro()
+
+#===============================================================================
+# Set MSVC static runtime library policy
+# NOTE: Since this sets a cmake policy for ensuring Windows compilation uses static runtime,
+# this policy MUST be set before any projects are declared for it to take effect.
+macro(set_msvc_static_runtime_policy)
+    # Enforce use of static runtime (avoids target machine needing msvcrt installed).
+    # Must set policy CMP0091 before any projects are declared (see https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html#cmake-msvc-runtime-library)
+    cmake_policy(SET CMP0091 NEW)
+    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+endmacro()
+
+#===============================================================================
+# Locate Spectre mitigated static C runtime on Windows. Sets the `SPECTRE_LIBS`
+# variable for use in later calls to `target_link_libraries`.
+# NOTE: This sets a cmake policy for ensuring Windows compilation uses static runtime.
+# this policy MUST be set before any projects are declared for it to take effect.
+macro(locate_win32_spectre_static_runtime)
+    if (WIN32)
+        # Locate the vswhere application, which will provide paths to any installed Visual Studio instances.
+        # By invoking it with "-find **/lib/spectre/x64" we will find any Spectre mitigated libaries that
+        # have been installed.
+        find_program(_vswhere_tool
+            NAMES vswhere
+            PATHS "$ENV{ProgramFiles\(x86\)}/Microsoft Visual Studio/Installer")
+        if (NOT ${vswhere})
+            message(FATAL_ERROR "Could not locate vswhere.exe - unable to source vc redistributable")
+        endif()
+        execute_process(
+            COMMAND "${_vswhere_tool}" -latest -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -find **/14.29.*/**/lib/spectre/x64 -sort
```

IIRC, when adapting for VS2022, I had to do the following change:
```suggestion
            COMMAND "${_vswhere_tool}" -latest -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -find **/14.*.*/**/lib/spectre/x64 -sort
```

---

Unfortunately the VS2022 version binaries didn't pass the BinSkim checks. You can see that in the log from the SDL pipeline [here](https://dev.azure.com/ms-quantum/Quantum%20Program/_build/results?buildId=220094&view=logs&j=663ee715-dcc5-5b8a-8432-471131753a80&t=c27f7e01-8477-5121-dfc4-3699acde542a&l=137), and the lines in the build log [here](https://dev.azure.com/ms-quantum/Quantum%20Program/_build/results?buildId=220094&view=logs&j=01472334-6cb8-5a65-5848-8350d96b6b87&t=407b8715-ddab-5d3e-3fdb-ca4f6740b4ba&l=104) showing that the selected runtime was the `14.32.31326` with Spectre mitigations. It's not clear to me why the latest binaries don't pass the checks even though they come from the spectre path, but I do know that if the `14.16` or `14.29` versions are chose those pass the checks, and both of those are installed in the build pipeline by default (the build pipeline has MSVC v143, v142, and v141 installed in the VS2022 folder). This means building locally will require installing the Spectre mitigated v142 components, which are optional installs on VS2022.